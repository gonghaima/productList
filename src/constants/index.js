import keyMirror from 'fbjs/lib/keyMirror';

/**
 * @namespace Constants
 * @desc App constants
 */

/**
 * @constant {Object} ActionTypes
 * @memberof Constants
 */
export const ActionTypes = keyMirror({
  SWITCH_MENU: undefined,
  EXCEPTION: undefined,
  PRODUCT_GET_REPOS: undefined,
  PRODUCT_GET_REPOS_SUCCESS: undefined,
  PRODUCT_GET_REPOS_FAILURE: undefined,
  SELECTION_CHANGE: undefined,
  PAGE_CHANGE: undefined,
});

/**
 * @constant {Object} STATUS
 * @memberof Constants
 */
export const STATUS = {
  IDLE: 'idle',
  RUNNING: 'running',
  READY: 'ready',
  SUCCESS: 'success',
  ERROR: 'error',
};
