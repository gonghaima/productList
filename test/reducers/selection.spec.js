import reducer from 'reducers/Selection';
import { ActionTypes } from 'constants/index';

describe('Selection', () => {
  it('should return the initial state', () => {
    expect(reducer.selection(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.SELECTION_CHANGE}`, () => {
    expect(
      reducer.selection(undefined, {
        type: ActionTypes.SELECTION_CHANGE,
        payload: { q: 'react' },
      }),
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PAGE_CHANGE}`, () => {
    expect(
      reducer.selection(undefined, {
        type: ActionTypes.PAGE_CHANGE,
        payload: {},
      }),
    ).toMatchSnapshot();
  });
});
