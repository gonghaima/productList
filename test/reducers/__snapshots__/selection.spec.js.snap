// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection should handle PAGE_CHANGE 1`] = `
Object {
  "currentPage": undefined,
  "itemPerPage": 4,
  "selectionItems": Array [
    4,
    8,
    10,
    20,
    40,
  ],
}
`;

exports[`Selection should handle SELECTION_CHANGE 1`] = `
Object {
  "currentPage": 0,
  "itemPerPage": undefined,
  "selectionItems": Array [
    4,
    8,
    10,
    20,
    40,
  ],
}
`;

exports[`Selection should return the initial state 1`] = `
Object {
  "currentPage": 0,
  "itemPerPage": 4,
  "selectionItems": Array [
    4,
    8,
    10,
    20,
    40,
  ],
}
`;
